signer/threshold_ed25519_localsoftsign_signer.go

```diff
@@ -0,0 +1,369 @@
+package signer
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+
+	tmCryptoEd25519 "github.com/tendermint/tendermint/crypto/ed25519"
+	tmJson "github.com/tendermint/tendermint/libs/json"
+	"gitlab.com/unit410/edwards25519"
+	tsed25519 "gitlab.com/unit410/threshold-ed25519/pkg"
+)
+
+// LocalSoftsignThresholdEd25519Signature implements the interface and signs the message for each local signer.
+// LocalSoftSignThresholdEd25519Signature is the implementation of a soft sign signer at the local level.
+type LocalSoftSignThresholdEd25519Signature struct {
+	PubKeyBytes []byte
+	Key         CosignerKey
+	RsaKey      rsa.PrivateKey
+	// Total signers
+	Total     uint8
+	Threshold uint8
+	// Height, Round, Step, Timestamp -> metadata
+	HrsMeta map[HRSTKey]HrsMetadata
+}
+
+// Inistiates the LocalSoftSignThresholdEd25519SignatureConfig struct.
+func NewLocalSoftSignThresholdEd25519SignatureConfig(
```

Comment:
This method and `NewThresholdEd25519Signature` on line 53 can be combined to a single function that returns `ThresholdEd25519Signature`. It could be named something like `NewSoftSignThresholdEd25519Signature`. The intermediate config type `LocalSoftSignThresholdEd25519SignatureConfig` can then be removed.